### PR TITLE
feat: add static search

### PR DESCRIPTION
**Pull request by Revathyvenugopal162**
static fuse search and remove meilisearch

### CODE REVIEW

File: src/ansys_sphinx_theme/search/fuse_search.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    all_docs = app.env.found_docs
+    search_index_list = []
+
+    for doc in all_docs:
+        doc_search = GetSearchIndex(doc, app)
+        doc_search.iterate_through_docs()
+        search_index_list.extend(doc_search.get_search_index())
+
+    with open(app.builder.outdir / "search.json", "w", encoding="utf-8") as f:  # noqa: PTH123
+        json.dump(search_index_list, f, ensure_ascii=False, indent=4)
```
Comment (jorgepiloto) [inline src/ansys_sphinx_theme/search/fuse_search.py:112]: Just fixing the path rule:

```suggestion
    search_index = app.builder.outdir / "search.json"
    with search_index.open("w", encoding="utf-8") as index_file:
        json.dump(indices, index_file, ensure_ascii=False, indent=4)
```

File: src/ansys_sphinx_theme/theme/ansys_sphinx_theme/components/search-field.html [hunk 2]
```diff
@@ -1,80 +1,149 @@
...
+<script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.6/require.min.js"></script>
+
+
+<!-- Script tag for search functionality -->
+ {% set theme_static_search = theme_static_search | default({}) %}
+  {% set threshold = theme_static_search.threshold | default(0.5) %}
+  {% set use_extended_search = theme_static_search.use_extended_search | default(true) %}
+  {% set should_sort = theme_static_search.should_sort | default(true) %}
+  {% set ignoreLocation = theme_static_search.ignoreLocation | default(false) %}
+  {% set limit = theme_static_search.docs_limit | default(20) %}
```
Comment (jorgepiloto) [inline src/ansys_sphinx_theme/theme/ansys_sphinx_theme/components/search-field.html:21]: Nice defaults!

File: src/ansys_sphinx_theme/search/fuse_search.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+"""Module to fuse search."""
+
+import json
+import re
+
+from docutils import nodes
+
+
+class GetSearchIndex:
```
Comment (jorgepiloto) [inline src/ansys_sphinx_theme/search/fuse_search.py:31]: Dropping the verb for this class:

```suggestion
class SearchIndex:
```

File: src/ansys_sphinx_theme/search/fuse_search.py [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+        Parameters
+        ----------
+        doc_name : str
+            Document name.
+        app : Sphinx
+            Sphinx application.
+        """
+        self._doc_name = doc_name
+        self.doc_path = self._doc_name + ".html"
```
Comment (jorgepiloto) [inline src/ansys_sphinx_theme/search/fuse_search.py:45]:
```suggestion
        self.doc_path = f"{self._doc_name}.html}"
```

File: src/ansys_sphinx_theme/search/fuse_search.py [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+    if not app.config.html_theme_options.get("static_search"):
+        return
+
+    all_docs = app.env.found_docs
+    search_index_list = []
+
+    for doc in all_docs:
+        doc_search = GetSearchIndex(doc, app)
+        doc_search.iterate_through_docs()
+        search_index_list.extend(doc_search.get_search_index())
```
Comment (jorgepiloto) [inline src/ansys_sphinx_theme/search/fuse_search.py:109]:
```suggestion
    for document in all_docs:
        search_index = SearchIndex(document, app)
        search_index_list.extend(search_index.indices)
```

File: src/ansys_sphinx_theme/search/fuse_search.py [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+        for sections in self.sections:
+            search_index = {
+                "objectID": self._doc_name,
+                "href": f"{self.doc_path}#{sections['section_anchor_id']}",
+                "title": self.doc_title,
+                "section": sections["section_title"],
+                "text": sections["section_text"],
+            }
+            search_index_list.append(search_index)
+        return search_index_list
```
Comment (jorgepiloto) [inline src/ansys_sphinx_theme/search/fuse_search.py:84]: Maybe, we could return a generator. This way, we remove allocating a list. Users can always convert:

```suggestion
    def get_search_index(self):
        """Get search index."""
        for sections in self.sections:
            search_index = {
                "objectID": self._doc_name,
                "href": f"{self.doc_path}#{sections['section_anchor_id']}",
                "title": self.doc_title,
                "section": sections["section_title"],
                "text": sections["section_text"],
            }
            yield search_index
```

File: src/ansys_sphinx_theme/theme/ansys_sphinx_theme/components/search-field.html [hunk 7]
```diff
@@ -1,80 +1,149 @@
...
-      .form-control:focus,.form-control:focus-visible,.form-control {
-        width: 490px;
+</form>
+
+<!-- Add this div to show the search results -->
+<div id="results" class="results"></div>
+
+<link href="{{ pathto('_static/css/fuse-search.css', 1) }}" rel="stylesheet" />
+
+<script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.6/require.min.js"></script>
```
Comment (jorgepiloto) [inline src/ansys_sphinx_theme/theme/ansys_sphinx_theme/components/search-field.html:12]: What is this script needed for?

File: src/ansys_sphinx_theme/theme/ansys_sphinx_theme/components/search-field.html [hunk 8]
```diff
@@ -1,80 +1,149 @@
...
+  });
+
+  require(['fuse'], function(Fuse) {
+    let fuseInstance;
+    let searchData = [];
+
+    function initializeFuse(data) {
+      const fuseOptions = {
+        keys: ['title', 'text'],
+        threshold: "{{ threshold }}",
```
Comment (jorgepiloto) [inline src/ansys_sphinx_theme/theme/ansys_sphinx_theme/components/search-field.html:36]: Are these options coming from the `conf.py`?

File: src/ansys_sphinx_theme/theme/ansys_sphinx_theme/components/search-field.html [hunk 9]
```diff
@@ -1,80 +1,149 @@
...
-  function checkMeiliSearchHost(url, callback) {
-    var xhr = new XMLHttpRequest();
-    xhr.open('HEAD', url, true);
-    xhr.onload = function () {
-      if (xhr.status === 404) {
-        callback(false);
+
+      //if query is empty,  dont show the div result container
+      if (query === '') {
+        resultsContainer.style.display = 'none';
```
Comment (jorgepiloto) [inline src/ansys_sphinx_theme/theme/ansys_sphinx_theme/components/search-field.html:61]: Nice!

```suggestion
      // If query is empty,  dont show the div result container
      if (query === '') {
        resultsContainer.style.display = 'none';
```

File: src/ansys_sphinx_theme/theme/ansys_sphinx_theme/components/search-field.html [hunk 10]
```diff
@@ -1,80 +1,149 @@
...
+    // Split query into words and escape special characters for each word
+    const words = query.trim().split(/\s+/);
+    const escapedWords = words.map(word => word.replace(/[-\/\\^$*+?.()|[\]{}]/g, '\\$&'));
+
+    // Create a regex to match any of the words
+    const regex = new RegExp(`(${escapedWords.join('|')})`, 'gi');
+
+    // Replace matched text with highlighted version
+    return text.replace(regex, '<span class="highlight">$1</span>');
+}
```
Comment (jorgepiloto) [inline src/ansys_sphinx_theme/theme/ansys_sphinx_theme/components/search-field.html:106]: Adding some indentation:

```suggestion
    function highlightTerms(text, query) {
      if (!query.trim()) return text;

      // Split query into words and escape special characters for each word
      const words = query.trim().split(/\s+/);
      const escapedWords = words.map(word => word.replace(/[-\/\\^$*+?.()|[\]{}]/g, '\\$&'));

      // Create a regex to match any of the words
      const regex = new RegExp(`(${escapedWords.join('|')})`, 'gi');

      // Replace matched text with highlighted version
      return text.replace(regex, '<span class="highlight">$1</span>');
}
```

File: src/ansys_sphinx_theme/theme/ansys_sphinx_theme/components/search-field.html (hunk 8 above)
Comment (Revathyvenugopal162) [reply src/ansys_sphinx_theme/theme/ansys_sphinx_theme/components/search-field.html:36]: yes, it is from conf.py -> html theme options - static_search dict

File: src/ansys_sphinx_theme/theme/ansys_sphinx_theme/components/search-field.html [hunk 11]
```diff
@@ -1,80 +1,149 @@
...
-{% endif %}
+
+    // Add event listener for Enter key press
+    searchBox.addEventListener("keydown", function(event) {
+      if (event.key === "Enter") {
+        const firstResult = document.querySelector('.result-item');
+        if (firstResult) {
+          // Navigate to the first result's URL
+          navigateToHref(firstResult.getAttribute('data-href'));
+          // window.location.href = firstResult.getAttribute('data-href');
```
Comment (jorgepiloto) [inline src/ansys_sphinx_theme/theme/ansys_sphinx_theme/components/search-field.html:128]:
```suggestion
```

**Review by jorgepiloto — CHANGES_REQUESTED** [review]
Just left some suggestions.

File: src/ansys_sphinx_theme/theme/ansys_sphinx_theme/components/search-field.html (hunk 8 above)
Comment (jorgepiloto) [reply src/ansys_sphinx_theme/theme/ansys_sphinx_theme/components/search-field.html:36]: Thanks!

File: src/ansys_sphinx_theme/theme/ansys_sphinx_theme/components/search-field.html (hunk 7 above)
Comment (Revathyvenugopal162) [reply src/ansys_sphinx_theme/theme/ansys_sphinx_theme/components/search-field.html:12]: At some point, the sphinx(may be pydata theme) was raising an error for not using require.js. It is a module loader https://requirejs.org/. We need to load fuse module.

Comment (jorgepiloto) [reply src/ansys_sphinx_theme/theme/ansys_sphinx_theme/components/search-field.html:12]: Thanks for the explanation.

File: src/ansys_sphinx_theme/theme/ansys_sphinx_theme/components/search-field.html [hunk 12]
```diff
@@ -13,6 +13,7 @@
 
 {% set theme_static_search = theme_static_search | default({}) %}
 {% set theme_limit = theme_static_search.limit | default(10) %}
+{% set start_search_at = theme_static_search.start_search_at | default(1) %}
```
Comment (jorgepiloto) [inline src/ansys_sphinx_theme/theme/ansys_sphinx_theme/components/search-field.html:16]: What about "min_chars_for_search"?

Comment (Revathyvenugopal162) [reply src/ansys_sphinx_theme/theme/ansys_sphinx_theme/components/search-field.html:16]: sure adding it.

File: doc/source/user-guide/options.rst [hunk 13]
```diff
@@ -108,92 +108,51 @@ If you want to hide all icons, use the ``show_icons`` Boolean variable:
...
-a child dictionary named ``use_meilisearch``is added to the ``html_theme_options``
-dictionary.
-
-This dictionary contains these keys, in the order given:
-
-#. ``host``: Host name of your MeiliSearch instance. If no value is provided,
-   the default public host for PyAnsys is used: ``https://backend.search.pyansys.com``
-   on port ``7700``. If added security is needed, you can use the ``os.getenv()`` function
-   to set the instance using an environment variable.
+Static Search Options
```
Comment (Revathyvenugopal162) [inline doc/source/user-guide/options.rst:111]:
```suggestion
Static search options
```

File: doc/source/user-guide/options.rst [hunk 14]
```diff
@@ -108,92 +108,51 @@ If you want to hide all icons, use the ``show_icons`` Boolean variable:
...
 
 
-Here is an example configuration of how to configure MeiliSearch in the ``conf.py`` file
-for the Ansys Sphinx Theme:
-
-.. code-block:: python
+.. note::
 
-    import os
+    To use the search bar in local documentation, you must open the documentation through a local server.
```
Comment (jorgepiloto) [inline doc/source/user-guide/options.rst:145]:
```suggestion
    Serve locally your documentation using the ``python -m http.server -d /path/to/docs/html/`` to have a live-preview of your search results. This method is compliant with the `CORS policy <https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS>`_ and allows to load the generated resource files containing the indices of your documentation.
```

File: src/ansys_sphinx_theme/__init__.py [hunk 15]
```diff
@@ -534,6 +514,21 @@ def build_quarto_cheatsheet(app: Sphinx):
     app.config.html_theme_options["cheatsheet"]["thumbnail"] = f"{output_dir}/{output_png}"
 
 
+def check_for_depreciated_theme_options(app: Sphinx):
```
Comment (jorgepiloto) [inline src/ansys_sphinx_theme/__init__.py:517]: Awesome! Thanks for adding this check.

**Review by jorgepiloto — CHANGES_REQUESTED** [review]
Just left some suggestions about the docs. Overall this looks superb!

**Review by jorgepiloto — APPROVED** [review]

**Comment by jorgepiloto** [comment]
Thanks for the big effort here, @Revathyvenugopal162

**Pull request merged**